### PR TITLE
Doc Improve. Replace timeseries_split unit-tests with a simple test

### PR DESCRIPTION
**Pull request by Sahil333**

### CODE REVIEW

File: examples/meta/src/evaluation/cross_validation_timeseries_split.sg [hunk 1]
```diff
@@ -6,16 +6,16 @@ RegressionLabels labels(f_labels)
 
 #![set parameters]
 int num_subsets = 5
-int min_subset_size = 4
+int future_steps = 4
```
Comment (karlnapf) [inline examples/meta/src/evaluation/cross_validation_timeseries_split.sg]: I'd suggest `future_offset`  but that is minor

File: examples/meta/src/evaluation/cross_validation_timeseries_split.sg [hunk 2]
```diff
@@ -6,16 +6,16 @@ RegressionLabels labels(f_labels)
...
+splitting.set_min_future_steps(future_steps)
 #![set parameters]
 
 #![build subsets]
 splitting.build_subsets()
 #![build subsets]
 
-#![generate subsets and inverse (aka test labels and train labels)]
-IntVector test_labels_indices = splitting.generate_subset_indices(1)
+#![generate subsets and inverse (aka vaildation set and train set)]
```
Comment (karlnapf) [inline examples/meta/src/evaluation/cross_validation_timeseries_split.sg]: This is a tag for the cookbooks, not free form comment. So pls make it super short (see other examples)

File: src/shogun/evaluation/TimeSeriesSplitting.cpp [hunk 3]
```diff
@@ -49,59 +49,60 @@ CTimeSeriesSplitting::CTimeSeriesSplitting(CLabels* labels, index_t num_subsets)
...
 void CTimeSeriesSplitting::init()
 {
-	m_rng = sg_rand;
-	m_min_subset_size = 1;
+	m_min_future_steps = 1;
 }
 
 void CTimeSeriesSplitting::build_subsets()
 {
+	REQUIRE(m_num_subsets > 0, "Number of subsets should be greater than 0.");
```
Comment (karlnapf) [inline src/shogun/evaluation/TimeSeriesSplitting.cpp]: Missing newline

File: src/shogun/evaluation/TimeSeriesSplitting.cpp [hunk 4]
```diff
@@ -49,59 +49,60 @@ CTimeSeriesSplitting::CTimeSeriesSplitting(CLabels* labels, index_t num_subsets)
...
-			split_index = indices.vlen - m_min_subset_size;
+		if (i == m_num_subsets - 1)
+			split_index = indices.vlen - m_min_future_steps;
 		else
-			split_index = (i + 1) * (indices.vlen / num_subsets);
+			split_index = (i + 1) * (indices.vlen / m_num_subsets);
 
-		/* filling current with indices on right end  */
+		/* filling current with indices on right side */
 		for (auto k = split_index; k < indices.vlen; ++k)
```
Comment (karlnapf) [inline src/shogun/evaluation/TimeSeriesSplitting.cpp]: `auto k : Range(indices.vlen)` but it is minor

File: src/shogun/evaluation/TimeSeriesSplitting.cpp [hunk 5]
```diff
@@ -49,59 +49,60 @@ CTimeSeriesSplitting::CTimeSeriesSplitting(CLabels* labels, index_t num_subsets)
...
+	REQUIRE(m_num_subsets > 0, "Number of subsets should be greater than 0.");
+
 	index_t num_labels = m_labels->get_num_labels();
 
-	/* min_size should be less than difference between number of labels
-	 * and split index of second last split */
-	REQUIRE(min_size > 0, "Minimum subset size should be atleast 1.")
+	/* future_steps should be less than the difference between number of labels
+	 * and split index of second last fold */
+	index_t future_steps_upperbound =
```
Comment (karlnapf) [inline src/shogun/evaluation/TimeSeriesSplitting.cpp]: max_future_steps
or better max_future_offset

File: src/shogun/evaluation/TimeSeriesSplitting.h [hunk 6]
```diff
@@ -40,55 +40,60 @@
 namespace shogun
 {
 	class CLabels;
-
-	/** @brief Implements a timeseries splitting strategy for cross-validation.
-	 * The strategy builds a given number of subsets each filled with labels
-	 * indices greater than a split index. The split indices are \f$ c[N/K] \f$
+	/** @brief Implements a timeseries splitting strategy for cross-validation,
+	 * respecting time.
+	 * Each fold splits timeseries into train (subset_inverse) and validation
```
Comment (karlnapf) [inline src/shogun/evaluation/TimeSeriesSplitting.h]: pls remove the (subset_inverse) here. This is documented in the base class
The newlines are awkwards, could you remove them?

File: src/shogun/evaluation/TimeSeriesSplitting.h [hunk 7]
```diff
@@ -40,55 +40,60 @@
...
+	 * (subset) set.
+	 * Train set contains indices less than a split index
+	 * and validation set contains rest of the indices.
+	 * The split indices are \f$ C*floor(N/K) \f$
```
Comment (karlnapf) [inline src/shogun/evaluation/TimeSeriesSplitting.h]: in latex math, you don't use * for multiplication but \cdot or just nothing

File: tests/unit/evaluation/SplittingStrategy_unittest.cc [hunk 8]
```diff
@@ -297,89 +298,63 @@ TEST(SplittingStrategy,LOO)
...
-		    new CTimeSeriesSplitting(labels, num_subsets);
+		num_labels = 15;
+		num_subsets = 3;
+		future_steps = 3;
 
-		splitting->set_min_subset_size(min_subset_size);
-		splitting->build_subsets();
-
-		for (index_t i = 0; i < num_subsets; ++i)
+		labels = new CRegressionLabels(num_labels);
```
Comment (karlnapf) [inline tests/unit/evaluation/SplittingStrategy_unittest.cc]: labels = some<CRegressionLabels>(num_labels); and then you dont have to SG_UNREF later

File: tests/unit/evaluation/SplittingStrategy_unittest.cc [hunk 9]
```diff
@@ -297,89 +298,63 @@ TEST(SplittingStrategy,LOO)
...
 
-			EXPECT_EQ(subset.vlen + inverse.vlen, num_labels);
-		}
+	train_indices = splitting->generate_subset_inverse(2);
 
-		/* clean up */
-		SG_UNREF(splitting);
+	for (auto i : range(12))
+	{
+		EXPECT_EQ(train_indices[i], i);
```
Comment (karlnapf) [inline tests/unit/evaluation/SplittingStrategy_unittest.cc]: yes this is a better test in fact!

**Comment by vigsterkr** [comment]
@Sahil333 any update on this?

**Pull request closed without merging**